### PR TITLE
chore: Audit messages.rs

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

Optimize Vec with reference, clean up unused/dup methods, remove unwrap as much as possible
## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/crypto.rs [hunk 1]
```diff
@@ -1302,29 +1302,29 @@ impl<const STRONG_THRESHOLD: bool> AuthoritySignInfoTrait
 
 impl<const STRONG_THRESHOLD: bool> AuthorityQuorumSignInfo<STRONG_THRESHOLD> {
     pub fn new_from_auth_sign_infos(
-        auth_sign_infos: Vec<AuthoritySignInfo>,
+        auth_sign_infos: &[AuthoritySignInfo],
         committee: &Committee,
     ) -> SuiResult<Self> {
-        fp_ensure!(
-            auth_sign_infos.iter().all(|a| a.epoch == committee.epoch),
```
Comment (joyqvq) [inline crates/sui-types/src/crypto.rs:1309]: remove this iter to be combined with the stake aggregator.

File: crates/sui-types/src/message_envelope.rs [hunk 2]
```diff
@@ -271,11 +269,6 @@ where
 }
 
 impl<T: Message, S> VerifiedEnvelope<T, S> {
-    /// This API should only be called when the input is already verified.
-    pub fn new_from_verified(inner: Envelope<T, S>) -> Self {
-        Self(TrustedEnvelope(inner), NoSer)
-    }
```
Comment (joyqvq) [inline crates/sui-types/src/message_envelope.rs:277]: this is a dup to new_unchecked few lines below

**Review by kchalkias — COMMENTED** [review]
2023-04-04T03:27:42.638399Z  WARN sui_core::stake_aggregator: crates/sui-core/src/stake_aggregator.rs:156: Bad stake from validator: InvalidSignature { error: "General cryptographic error" } name=k#99f25ef6..
test authority_aggregator::authority_aggregator_tests::test_byzantine_authority_sig_aggregation ... FAILED

**Review by kchalkias — COMMENTED** [review]
@joyqvq thanks for the PR, do we need anything else to unblock it apart from conflict resolution and the previously mentioned error?

File: crates/sui-types/src/crypto.rs [hunk 3]
```diff
@@ -1318,29 +1318,29 @@ impl<const STRONG_THRESHOLD: bool> AuthoritySignInfoTrait
...
-            SuiError::InvalidSignature {
-                error: "All signatures must be from the same epoch as the committee".to_string()
-            }
-        );
-        let total_stake: StakeUnit = auth_sign_infos
-            .iter()
-            .map(|a| committee.weight(&a.authority))
```
Comment (joyqvq) [inline crates/sui-types/src/crypto.rs:1332]: combine two round of for loop into one

**Comment by joyqvq** [comment]
sorry for the delay on this - PR is ready for review

File: crates/sui-types/src/crypto.rs [hunk 4]
```diff
@@ -1318,29 +1318,29 @@ impl<const STRONG_THRESHOLD: bool> AuthoritySignInfoTrait
 
 impl<const STRONG_THRESHOLD: bool> AuthorityQuorumSignInfo<STRONG_THRESHOLD> {
     pub fn new_from_auth_sign_infos(
```
Comment (benr-ml) [inline crates/sui-types/src/crypto.rs:1320]: One thing that is strange about this function is that it trusts the caller to verify that all sign_infos came from different authorities, but not that the quorum threshold was reached. Not sure we want the second assumption, wdyt?

File: crates/sui-types/src/error.rs [hunk 5]
```diff
@@ -657,6 +657,7 @@ impl SuiError {
                 (false, true)
             }
             SuiError::ObjectLockConflict { .. } => (false, true),
+            SuiError::CertificateRequiresQuorum => (true, true),
```
Comment (benr-ml) [inline crates/sui-types/src/error.rs:660]: if we get to this error, it means that something is broken in `stake_aggregator`, no?

**Comment by joyqvq** [comment]
will rebase and open a new PR

**Pull request closed without merging**